controllers/kubeadmcontrolplane_controller.go

```diff
@@ -115,6 +117,13 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, log logr.
 		return ctrl.Result{}, err
 	}
 
+	log.Info("Pausing control plane machine health check", "MachineHealthCheck", cpMachineHealthCheckName(kcp.Name))
+	mhc := &clusterv1.MachineHealthCheck{}
+	result, err := r.pauseMachineHealthCheck(ctx, mhc, kcp.Name)
+	if err != nil {
+		return result, err
+	}
```

Comment:
Instead of pausing and unpausing every reconcile loop, can we pause whenever we are creating the CPU (right above line 156) and also unpause after deleting the CPU after line 133 (you already have it after line 173 which is good)?
Same thing for MDs